src/db/blob_db_manager.rs

```diff
@@ -6,7 +6,7 @@ use crate::types::{Blob, BlockData, IndexerMetadata, StdError, TransactionData};
 pub trait DBManager {
     type Options;
 
-    async fn new(connection_uri: &String, db_name: &String) -> Result<Self, StdError>
+    async fn new(connection_uri: &str, db_name: &str) -> Result<Self, StdError>
```

Comment:
a note here: generally better to use base types to avoid unnecessary allocations and less indirection: https://rust-unofficial.github.io/patterns/idioms/coercion-arguments.html

Reply:
Interesting,, I'll take note of this